Mathlib/Util/Notation3.lean

```diff
@@ -250,7 +258,22 @@ partial def exprToMatcher (boundFVars : Std.HashMap FVarId Name)
   match e with
   | .mvar .. => return ([], ← `(pure))
   | .const n _ => return ([.app n 0], ← ``(matchExpr (Expr.isConstOf · $(quote n))))
-  | .sort .. => return ([.other `sort], ← ``(matchExpr Expr.isSort))
+  | .sort u =>
+    /-
+    We should try being more accurate here.
+    Prop / Type / Type _ / Sort _ is at least an OK approximation.
+    We mimic the core Sort delaborator `Lean.PrettyPrinter.Delaborator.delabSort`.
+    -/
+    let matcher ←
+      if u.isZero then
+        ``(matchExpr Expr.isProp)
+      else if e.isType0 then
```

Comment:
I guess you could also use
```suggestion
      else if u matches .succ .zero then
```
here to avoid matching a second time, but I guess that's not really worth worrying about.

Reply:
Yeah, ideally there would be some function on universe levels to see if it's levelOne (I suppose I could have done `u == levelOne`).

I'm hoping that the improvement in this PR is relatively temporary, and that eventually there will be a more precise universe level checker. Because of that, let's leave it as-is for now.